pkg/restic/common.go

```diff
@@ -241,14 +241,17 @@ func GetSnapshotsInBackup(backup *velerov1api.Backup, podVolumeBackupLister vele
 // encryption key for the given repo and returns its path. The
 // caller should generally call os.Remove() to remove the file
 // when done with it.
-func TempCredentialsFile(secretLister corev1listers.SecretLister, veleroNamespace, repoName string, fs filesystem.Interface) (string, error) {
-	secretGetter := NewListerSecretGetter(secretLister)
-
+func TempCredentialsFile(client kbclient.Client, veleroNamespace, repoName string, fs filesystem.Interface) (string, error) {
 	// For now, all restic repos share the same key so we don't need the repoName to fetch it.
 	// When we move to full-backup encryption, we'll likely have a separate key per restic repo
-	// (all within the Velero server's namespace) so GetRepositoryKey will need to take the repo
-	// name as an argument as well.
-	repoKey, err := GetRepositoryKey(secretGetter, veleroNamespace)
+	// (all within the Velero server's namespace) so repoKeySelector will need to select the key
+	// for that repo.
+	repoKeySelector := &corev1api.SecretKeySelector{
```

Comment:
Let's add a build for the SecretKeySelector to get more consistency in the code base.

Reply:
Just to clarify, do you mean a `SecretKeySelector` builder in `pkg/builder`

Reply:
Yep, sorry, I realize I didn't complete the word.

Reply:
I've added the builder and used in the new code, but I also updated the CLI changes to use it. Hopefully that was part of what you intended but if not, let me know and I'll revert those changes :)